source/Octopus.Tentacle.Tests.Integration/Support/TentacleConfigurationsAttribute.cs

```diff
@@ -0,0 +1,131 @@
+            {
+                versions.AddRange(new[]
+                {
+                    TentacleVersions.Current,
+                    TentacleVersions.v5_0_15_LastOfVersion5,
+                    TentacleVersions.v6_3_417_LastWithScriptServiceV1Only,
+                    TentacleVersions.v7_0_1_ScriptServiceV2Added
+                });
+            }
+
+            if (testCapabilitiesInterestingVersions)
+            {
+                versions.AddRange(new[]
+                {
+                    TentacleVersions.Current,
+                    TentacleVersions.v5_0_4_FirstLinuxRelease,
+                    TentacleVersions.v5_0_12_AutofacServiceFactoryIsInShared,
+                    TentacleVersions.v6_3_417_LastWithScriptServiceV1Only, // the autofac service is in tentacle, but tentacle does not have the capabilities service.
+                    TentacleVersions.v7_0_1_ScriptServiceV2Added
+                });
+            }
+
+            if (testScriptIsolationLevel)
+            {
+                versions.AddRange(new[]
+                {
+                    TentacleVersions.Current,
+                    TentacleVersions.v6_3_417_LastWithScriptServiceV1Only,
+                    TentacleVersions.v7_0_1_ScriptServiceV2Added // Testing against v1 and v2 script services
+                });
+            }
+
+            if (testNoCapabilitiesServiceVersions)
+            {
+                versions.AddRange(new []
+                {
+                    TentacleVersions.v6_3_451_NoCapabilitiesService
+                });
+            }
+
+            var testCases =
+                from tentacleType in tentacleTypes
+                from halibutType in halibutTypes
+                from version in versions.Distinct()
+                select new TentacleConfigurationTestCase(
+                    tentacleType,
+                    halibutType,
+                    version);
+
+            if (additionalParameterTypes.Length == 0)
+            {
+                return testCases.GetEnumerator();
+            }
+            
+            return CombineTestCasesWithAdditionalParameters(testCases, additionalParameterTypes);
+        }
+
+        private static IEnumerator CombineTestCasesWithAdditionalParameters(IEnumerable<TentacleConfigurationTestCase> testCases, object[] additionalEnumerables)
+        {
+            AllCombinations enums = new AllCombinations();
+            var additionalEnums = additionalEnumerables.Cast<Type>().ToArray();
+
+            for (int i = 0; i < additionalEnums.Length; i++)
+            {
+                var additionalEnum = additionalEnums[i];
+                if (additionalEnum.IsEnum)
+                {
+                    enums.And(Enum.GetValues(additionalEnums[i]));
+                }
+                else if (typeof(IEnumerable).IsAssignableFrom(additionalEnum))
+                {
+                    enums.And(ValuesOf.CreateValues(additionalEnum));
+                }
```

Comment:
Nit: Should we throw an exception if we aren't an enum or Enumerable? (or an enumerable with no values, based on previous comment 😁 )